src/main/java/gregtech/api/unification/material/Material.java

```diff
@@ -110,6 +130,14 @@ public Material setFormula(String formula, boolean withFormatting) {
         return this;
     }
 
+    @ZenMethod
+    public Material setComponents(MaterialStack... components) {
+        this.materialInfo.setComponents(components);
+        this.chemicalFormula = null;
```

Comment:
There is no point in setting the formula to null here, is there?

Reply:
If the formula isn't set `null`, then it won't actually calculate the new formula. https://github.com/GregTechCEu/GregTech/blob/586828f5697fbc92ec0613f1f1b6f5dd6a529132/src/main/java/gregtech/api/unification/material/Material.java#L77

Reply:
funny